P2022/tests/header/3way.hpp

```diff
@@ -0,0 +1,68 @@
+
+
+#include <iterator>
+#include <concepts>
+
+namespace std::ranges
+{
+    template<
+        input_iterator I1,
+        input_iterator I2,
+        class Comp,
+        class Proj1,
+        class Proj2
+    >
+    using lexicographical_compare_three_way_result_t =
+        invoke_result_t<Comp, invoke_result_t<Proj1, I1>, invoke_result_t<Proj2, I2>>;
```

Comment:
This line simply not compiles:
```suggestion
invoke_result_t<Comp, projected<I1,Proj1>, projected<I2,Proj2>>;
```